processing/src/main/java/io/druid/segment/incremental/OnheapIncrementalIndex.java

```diff
@@ -64,14 +73,39 @@
       boolean reportParseExceptions,
       boolean concurrentEventAdd,
       boolean sortFacts,
-      int maxRowCount
+      int maxRowCount,
+      long maxBytesInMemory
   )
   {
     super(incrementalIndexSchema, deserializeComplexMetrics, reportParseExceptions, concurrentEventAdd);
     this.maxRowCount = maxRowCount;
-
+    this.maxBytesInMemory = maxBytesInMemory == 0 ? defaultMaxBytesInMemory : maxBytesInMemory;
     this.facts = incrementalIndexSchema.isRollup() ? new RollupFactsHolder(sortFacts, dimsComparator(), getDimensions())
                                                    : new PlainFactsHolder(sortFacts);
+    maxBytesPerRowForAggregators = getMaxBytesPerRowForAggregators(incrementalIndexSchema);
```

Comment:
Conventionally, we include "this", like `this.maxBytesPerRowForAggregators =` in constructors when initializing fields. It doesn't matter for correctness, but it's conventional style in the codebase.

Reply:
done